xla/service/gpu/gpu_layout_assignment.cc

```diff
@@ -137,9 +140,21 @@ HeuristicLayoutAssignment(const HloInstruction* instr,
       instr->shape().tuple_shapes(0).dimensions_size() != 4) {
     return kAllNCHW;
   }
+#elif TENSORFLOW_USE_ROCM
```

Comment:
Are these ifdefs necessary? We're trying to minimise ifdefs from the codebase. Can you find a way to get rid of them? See https://github.com/openxla/xla/pull/10788 for inspiration.

Reply:
I've changed these to perform rocm vs cuda checks during runtime (with 'regular' if-else). Is that better? It also solves undefined variable error for CPU case.

Reply:
Thank you, much better.